Keda.Cosmosdb.Scaler/src/Services/CosmosDBExternalScaler.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            var resp = new GetMetricsResponse();
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScaledObjectRef.ScalerMetadata);
+                long workToBeDone = await GetEstimatedWork(trigger);
+
+                var metricName = request.ScaledObjectRef.ScalerMetadata["MetricName"];
+                resp.MetricValues.Add(new MetricValue
+                {
+                    MetricName = metricName,
+                    MetricValue_ = workToBeDone
+                });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error generating GetMetricsResponse");
+                throw ex;
+            }
+            return resp;
+        }
+
+        public CosmosDBTrigger CreateTriggerFromMetadata(MapField<string, string> scalerMetadata)
+        {
+            var lease = new CosmosDBLease()
+            {
+                LeasesCosmosDBConnectionString = scalerMetadata[CosmosDBLeaseMetadata.LeasesCosmosDBConnectionString],
+                LeaseDatabaseName = scalerMetadata[CosmosDBLeaseMetadata.LeaseDatabaseName],
+                LeaseCollectionName = scalerMetadata[CosmosDBLeaseMetadata.LeaseCollectionName]
+            };
+
+            var trigger = new CosmosDBTrigger()
+            {
+                CollectionName = scalerMetadata[CosmosDBTriggerMetadata.CollectionName],
+                DatabaseName = scalerMetadata[CosmosDBTriggerMetadata.DatabaseName],
+                CosmosDBConnectionString = scalerMetadata[CosmosDBTriggerMetadata.CosmosDBConnectionString],
+                AccountName = string.Empty,
+                Lease = lease
+            };
+
+            if (scalerMetadata.TryGetValue(CosmosDBTriggerMetadata.AccountName, out string accountName))
+            {
+                trigger.AccountName = accountName;
+            }
+
+            return trigger;
+        }
+
+        private async Task<long> GetEstimatedWork (CosmosDBTrigger trigger)
+        {
+            try
+            {
+                var builder = CreateChangeFeedProcessorBuilder(trigger);
+                var workEstimator = await builder.BuildEstimatorAsync();
+                long estimatedRemainingWork = await workEstimator.GetEstimatedRemainingWork();
+                return estimatedRemainingWork;
+            }
+            catch(Exception ex)
+            {
+                _logger.LogError(ex, "Error getting estimated work from host");
+                throw ex;
+            }
+        }
+
+        private static ChangeFeedProcessorBuilder CreateChangeFeedProcessorBuilder(
+            CosmosDBTrigger trigger)
+        {
+            CosmosDBConnectionString triggerConnection = new CosmosDBConnectionString(trigger.CosmosDBConnectionString);
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo

```

Comment:
https://github.com/Azure/azure-cosmos-dotnet-v3 is the latest version for Cosmos - please check if we can use the latest instead of older pacakges :`Microsoft.Azure.DocumentDB*`

Reply:
We definately should do that